drivers/storage/libstorage/libstorage_client_xcli.go

```diff
@@ -19,56 +19,57 @@ import (
 
 func (c *client) Supported(
 	ctx types.Context,
-	opts types.Store) (bool, error) {
+	opts types.Store) (types.LSXSupportedOp, error) {
 
 	if c.isController() {
-		return false, utils.NewUnsupportedForClientTypeError(
+		return 0, utils.NewUnsupportedForClientTypeError(
 			c.clientType, "Supported")
 	}
 
 	ctx = context.RequireTX(ctx.Join(c.ctx))
 
 	serviceName, ok := context.ServiceName(ctx)
 	if !ok {
-		return false, goof.New("missing service name")
+		return 0, goof.New("missing service name")
 	}
 
 	si, err := c.getServiceInfo(serviceName)
 	if err != nil {
-		return false, err
+		return 0, err
 	}
 	driverName := strings.ToLower(si.Driver.Name)
 
 	// check to see if the driver's executor is supported on this host
 	if ok := c.supportedCache.IsSet(driverName); ok {
-		return c.supportedCache.GetBool(driverName), nil
+		return c.supportedCache.GetLSXSupported(driverName), nil
 	}
 
 	out, err := c.runExecutor(ctx, driverName, types.LSXCmdSupported)
 	if err != nil {
 		if err == types.ErrNotImplemented {
 			ctx.WithField("serviceDriver", driverName).Warn(
 				"supported cmd not implemented")
-			c.supportedCache.Set(driverName, true)
+			c.supportedCache.Set(driverName, types.LSXOpAllNoMount)
```

Comment:
Hi @codenrhoden,

Pretty much :) It's back-compat. The only reason I'm still including it is the outstanding driver PRs there are. I don't know if they've all implemented `Supported` or not. I had originally used this PR to move the `Supported` function to the core type, but as I said, outstanding drivers made me think twice.